code/game/machinery/hologram.dm

```diff
@@ -90,182 +226,174 @@ var/list/holopads = list()
 	This may change in the future but for now will suffice.*/
 	if(user.eyeobj.loc != src.loc)//Set client eye on the object if it's not already.
 		user.eyeobj.setLoc(get_turf(src))
-	else if(!hologram)//If there is no hologram, possibly make one.
-		activate_holo(user, 0)
-	else if(master == user)//If there is a hologram, remove it. But only if the user is the master. Otherwise do nothing.
-		clear_holo()
-	return
-
-/obj/machinery/hologram/holopad/proc/activate_holo(mob/living/silicon/ai/user, var/force = 0)
-	if(!force && user.eyeobj.loc != src.loc) // allows holopads to pass off holograms to the next holopad in the chain
+	else if(!masters[user])//If there is no hologram, possibly make one.
+		activate_holo(user, 1)
+	else//If there is a hologram, remove it.
+		clear_holo(user)
+
+/obj/machinery/hologram/holopad/process()
+	for(var/I in masters)
+		var/mob/living/master = I
+		var/mob/living/silicon/ai/AI = master
+		if(!istype(AI))
+			AI = null
+
+		if(!qdeleted(master) && !master.incapacitated() && master.client && (!AI || AI.eyeobj))//If there is an AI attached, it's not incapacitated, it has a client, and the client eye is centered on the projector.
+			if(!(stat & NOPOWER))//If the  machine has power
+				if(AI)	//ais are range based
+					if(HOLOPAD_MODE == RANGE_BASED)
+						if(get_dist(AI.eyeobj, src) <= holo_range)
+							continue
+					else if(HOLOPAD_MODE == AREA_BASED)
+						var/area/holo_area = get_area(src)
+						var/area/eye_area = get_area(AI.eyeobj)
+
+						if(eye_area != holo_area)
+							clear_holo(master)
+							return 1
+
+					var/obj/machinery/hologram/holopad/pad_close = get_closest_atom(/obj/machinery/hologram/holopad, holopads, AI.eyeobj)
+					if(get_dist(pad_close, AI.eyeobj) <= pad_close.holo_range)
+						unset_holo(AI)
+						pad_close.activate_holo(AI, 1)
+				else
+					continue
+		clear_holo(master)//If not, we want to get rid of the hologram.

```

Comment:
You could remove the unset_holo above and move this to just after the HOLOPAD_MODE checks so that:
1. the comment makes sense
2. it clears the old hologram when they go out of range and then creates a new one if they are within range of another pad that is linked to the new pad.